File: MosPosudit.UI/shared/lib/dtos/order/order_update_request.dart
class OrderUpdateRequest {
  final DateTime? startDate;
  final DateTime? endDate;
  final num? totalAmount;
  final bool? isReturned;
  final DateTime? returnDate;

  OrderUpdateRequest({
    this.startDate,
    this.endDate,
    this.totalAmount,
    this.isReturned,
    this.returnDate,
  });

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> json = {};
    if (startDate != null) json['startDate'] = startDate!.toIso8601String();
    if (endDate != null) json['endDate'] = endDate!.toIso8601String();
    if (totalAmount != null) json['totalAmount'] = totalAmount;
    if (isReturned != null) json['isReturned'] = isReturned;
    if (returnDate != null) json['returnDate'] = returnDate!.toIso8601String();
    return json;
  }
}

File: MosPosudit.Model/Requests/Order/OrderUpdateRequest.cs
using System.ComponentModel.DataAnnotations;

namespace MosPosudit.Model.Requests.Order
{
    public class OrderUpdateRequest
    {
        public DateTime? StartDate { get; set; }

        public DateTime? EndDate { get; set; }

        [Range(0, 100000, ErrorMessage = "Total amount must be between 0 and 100000")]
        public decimal? TotalAmount { get; set; }

        public bool? IsReturned { get; set; }

        public DateTime? ReturnDate { get; set; }
    }
}

